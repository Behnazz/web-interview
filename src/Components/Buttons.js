import React from 'react'
import PropTypes from 'prop-types'

const Button = props => {
  const { selectedButton, handleClick, title } = props
  return (
    <button
      className={
        selectedButton === title ? 'appointment-button' : 'appointment-button1'
      }
      onClick={e => handleClick(title)}
    >
      {title}
    </button>
  )
}

Button.propTypes = {
  selectedButton: PropTypes.string,
  handleClick: PropTypes.func,
  title: PropTypes.string,
}
export default Button
