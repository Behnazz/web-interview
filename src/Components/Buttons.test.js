import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Buttons from './Buttons'
import { toMatchDiffSnapshot } from 'snapshot-diff'

expect.extend({ toMatchDiffSnapshot })

afterEach(cleanup)

it('match the snapshot', () => {
  const { asFragment } = render(<Buttons />)
  expect(asFragment()).toMatchSnapshot()
})

it('check the text of the buttons component', () => {
  const { getByText } = render(<Buttons title="Gp" />)
  expect(getByText('Gp')).toBeInTheDocument()
})

it('check click once', () => {
  const mockClick = jest.fn()
  const { getByText } = render(<Buttons title="Gp" handleClick={mockClick} />)
  fireEvent.click(getByText('Gp'))
  expect(mockClick).toHaveBeenCalledTimes(1)
})

it('change the color of the buttons', () => {
  const { asFragment } = render(<Buttons title="Gp" selectedButton="Gp" />)
  const firstRender = asFragment()
  expect(asFragment()).toMatchSnapshot()
  render(<Buttons title="Gp" selectedButton="" />)
  expect(firstRender).toMatchDiffSnapshot(asFragment())
})
