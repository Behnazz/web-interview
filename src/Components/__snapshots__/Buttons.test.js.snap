// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`change the color of the buttons 1`] = `
<DocumentFragment>
  <button
    class="button selected"
  >
    Gp
  </button>
</DocumentFragment>
`;

exports[`change the color of the buttons 2`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`match the snapshot 1`] = `
<DocumentFragment>
  <button
    class="button selected"
  />
</DocumentFragment>
`;
